### ISSUE TITLE
Update all packages

### ISSUE THREAD

**Issue by acgetchell** [conan-io/conan#4252]
`conan install --update --all` or something to allow updating all local packages to latest version.

Also submitted here: https://github.com/conan-io/wishlist/issues/214

**Comment by lasote** [comment]
The `conan install` command always operates targetting a `conanfile.py` or a reference to compute the graph and then (if specified) update the graph. So I don't see `conan install` as the right command. Can I ask what is your use case? Why is it not enough to update "by project"?

**Comment by acgetchell** [comment]
I'd like the ability to upgrade my Conan packages without having to go to Conan-Center or Bincrafters and manually look up what the latest version is.

The `conanfile.py` specifies the version explicitly, so it can't do that.

Instead, I'd like to be able to update installed packages, clean out outdated ones, and pin package versions if needed. I'd like to be able to inspect Conan's cache and clean it out as needed.

Basically, I want the `brew` interface:

https://docs.brew.sh/FAQ

Thanks for your response!

**Comment by lasote** [comment]
> I'd like the ability to upgrade my Conan packages without having to go to Conan-Center or Bincrafters and manually look up what the latest version is.

You don't need to go to the central repository to check the latest version of a package. The command `conan install --update` exists, but you need to do it in every project using Conan, to upgrade the packages in the dependency tree of the project.

That is the main difference with brew. With brew, the "consumer" of the installed bottles is the "system" or you as a user, but there is only one consumer. In the other hand with Conan, the consumer is any project or another package (a conanfile.txt or conanfile.py with requirements), so without checking all consumers, you cannot know which dependencies are used and which ones should be upgraded.
Also the Conan cache is exactly that, a cache, with brew I won't call it cache, it is an storage where the applications are installed.

> The conanfile.py specifies the version explicitly, so it can't do that.

You can use version ranges also, and the recipe will check in the remote if there is a new version available, but as I said, from a consumer package. We could develop a feature where we could iterate the local cache packages and check the remotes for updates, but it doesn't guarantee that the dependencies of your project will use the new ones, only if you are using a compatible version range, so for me it makes no much sense to do a "global upgrade" but upgrade each project individually. That is also because brew do not version the packages, so you get the only/latest available.

**Comment by acgetchell** [comment]
It's not immediately obvious to me how to use it in this way:

```
┌─[adam][hapkido][±][feature/refactor-moves ✓][~/CDT-plusplus]
└─▪ conan install --update
usage: conan install [-h] [-g GENERATOR] [-if INSTALL_FOLDER] [-m [MANIFESTS]]
                     [-mi [MANIFESTS_INTERACTIVE]] [-v [VERIFY]]
                     [--no-imports] [-j JSON] [-b [BUILD]] [-e ENV]
                     [-o OPTIONS] [-pr PROFILE] [-r REMOTE] [-s SETTINGS] [-u]
                     path_or_reference
conan install: error: the following arguments are required: path_or_reference
ERROR: Exiting with code: 2
```
What do I use for `path_or_reference`.

**Comment by lasote** [comment]
The `conanfile.py/conanfile.txt` of your project or a reference already installed in the cache. As I said, you need a "root" node to build the dep graph to be updated, you cannot just update the whole cache, unless you have a project that depends on all the packages in your local cache.

**Comment by acgetchell** [comment]
Okay, when I run that command I get:
```
┌─[adam][hapkido][±][feature/refactor-moves U:1 ✗][~/CDT-plusplus]
└─▪ conan install --update conanfile.py 
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=10.0
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

boost/1.69.0@conan/stable: Retrieving from remote 'conan-center'...
Downloading conanmanifest.txt
[==================================================] 125B/125B        
Downloading conanfile.py
[==================================================] 31.7KB/31.7KB    
Downloading conan_export.tgz
[==================================================] 602B/602B        
Decompressing conan_export.tgz: 100%|██████████| 602/602 [00:00<00:00, 217kB/s]
bzip2/1.0.6@conan/stable: Retrieving from remote 'conan-center'...
Downloading conanmanifest.txt
[==================================================] 163B/163B        
Downloading conanfile.py
[==================================================] 2.1KB/2.1KB      
Downloading conan_export.tgz
[==================================================] 747B/747B        
Decompressing conan_export.tgz: 100%|██████████| 747/747 [00:00<00:00, 552kB/s]
zlib/1.2.11@conan/stable: Retrieving from remote 'conan-center'...
Downloading conanmanifest.txt
[==================================================] 296B/296B        
Downloading conanfile.py
[==================================================] 11.6KB/11.6KB    
Downloading conan_export.tgz
[==================================================] 742B/742B        
Decompressing conan_export.tgz: 100%|██████████| 742/742 [00:00<00:00, 502kB/s]
docopt/0.6.2@conan/stable: WARN: Can't update, no package in remote
boost/1.69.0@conan/stable: WARN: Current package is older than remote upstream one
PROJECT: Installing /Users/adam/CDT-plusplus/conanfile.py
Requirements
    Catch2/2.5.0@catchorg/stable from 'conan-center' - Cache
    boost/1.69.0@conan/stable from 'conan-center' - Updated
    bzip2/1.0.6@conan/stable from 'conan-center' - Updated
    date/2.4.1@bincrafters/stable from 'bincrafters' - Cache
    docopt/0.6.2@conan/stable from 'conan-center' - Cache
    gsl_microsoft/2.0.0@bincrafters/stable from 'conan-center' - Cache
    zlib/1.2.11@conan/stable from 'conan-center' - Updated
Packages
    Catch2/2.5.0@catchorg/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    boost/1.69.0@conan/stable:411d816e9b8d6cb0bb1ef6e5be8b2a94b8b17c04 - Update
    bzip2/1.0.6@conan/stable:534dcc368c999e07e81f146b3466b8f656ef1f55 - Cache
    date/2.4.1@bincrafters/stable:60e5a79ffac27b15987d7fdaae6c763ec6371644 - Cache
    docopt/0.6.2@conan/stable:f8bda7f0751e4bc3beaa6c3b2eb02d455291c8a2 - Cache
    gsl_microsoft/2.0.0@bincrafters/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
    zlib/1.2.11@conan/stable:534dcc368c999e07e81f146b3466b8f656ef1f55 - Cache

Catch2/2.5.0@catchorg/stable: Already installed!
bzip2/1.0.6@conan/stable: Already installed!
date/2.4.1@bincrafters/stable: Already installed!
docopt/0.6.2@conan/stable: Already installed!
gsl_microsoft/2.0.0@bincrafters/stable: Already installed!
zlib/1.2.11@conan/stable: Already installed!
boost/1.69.0@conan/stable: Retrieving package 411d816e9b8d6cb0bb1ef6e5be8b2a94b8b17c04 from remote 'conan-center' 
Downloading conanmanifest.txt
[==================================================] 1.1MB/1.1MB      
Downloading conaninfo.txt
[==================================================] 2.5KB/2.5KB      
Downloading conan_package.tgz
[==================================================] 18.0MB/18.0MB    
Decompressing conan_package.tgz:   1%|          | 155k/17.1M [00:00<00:11, 1.59MDecompressing conan_package.tgz: 100%|██████████| 17.1M/17.1M [00:06<00:00, 2.81MB/s]
boost/1.69.0@conan/stable: Package installed 411d816e9b8d6cb0bb1ef6e5be8b2a94b8b17c04
boost/1.69.0@conan/stable: LIBRARIES: ['boost_wave', 'boost_container', 'boost_contract', 'boost_exception', 'boost_graph', 'boost_iostreams', 'boost_locale', 'boost_log', 'boost_program_options', 'boost_random', 'boost_regex', 'boost_wserialization', 'boost_serialization', 'boost_coroutine', 'boost_context', 'boost_timer', 'boost_thread', 'boost_chrono', 'boost_date_time', 'boost_atomic', 'boost_filesystem', 'boost_system', 'boost_type_erasure', 'boost_stacktrace_basic', 'boost_math_tr1', 'boost_math_c99f', 'boost_math_c99l', 'boost_log_setup', 'boost_stacktrace_addr2line', 'boost_stacktrace_noop', 'boost_math_c99', 'boost_math_tr1l', 'boost_math_tr1f', 'boost_unit_test_framework']
boost/1.69.0@conan/stable: Package folder: /Users/adam/.conan/data/boost/1.69.0/conan/stable/package/411d816e9b8d6cb0bb1ef6e5be8b2a94b8b17c04
PROJECT: Generator cmake created conanbuildinfo.cmake
PROJECT: Generator txt created conanbuildinfo.txt
PROJECT: Generated conaninfo.txt
PROJECT: Generated graphinfo
```
This generates `conanbuildinfo.txt` et. al but I still don't see how this updates my packages to the latest. I can't use `boost/1.Y.Z` in my `conanfile.py`, for example.

So I need to run this command, inspect `[full_requires]` from `conaninfo.txt`, and then update `conanfile.py/txt` accordingly?

**Comment by lasote** [comment]
boost/1.Y.Z is not a valid version range. That is the "hashed" value that conan uses internally to compute the package ID.

https://docs.conan.io/en/latest/mastering/version_ranges.html

**Comment by acgetchell** [comment]
Thanks, version range expressions work.

**Comment by bugproof** [comment]
Is there any way to update conanfile.txt versions with this command? Similar to how npm/yarn works

**Comment by memsharded** [comment]
Hi @bugproof 

No, right now there isn't. We are considering to add something, but so far not a priority, as adding it to the file is easy enough.
